src/main/java/bg/libapp/libraryapp/service/BookService.java

```diff
@@ -1,82 +1,167 @@
 package bg.libapp.libraryapp.service;
 
-import bg.libapp.libraryapp.exceptions.AuthorNotFoundException;
-import bg.libapp.libraryapp.exceptions.BookAlreadyAddedException;
-import bg.libapp.libraryapp.exceptions.BookNotFoundException;
-import bg.libapp.libraryapp.exceptions.GenreNotFoundException;
-import bg.libapp.libraryapp.model.dto.book.BookAddDTO;
-import bg.libapp.libraryapp.model.dto.book.BookViewDTO;
+import bg.libapp.libraryapp.event.BookAuditEvent;
+import bg.libapp.libraryapp.exceptions.*;
+import bg.libapp.libraryapp.model.dto.book.*;
 import bg.libapp.libraryapp.model.entity.Author;
 import bg.libapp.libraryapp.model.entity.Book;
 import bg.libapp.libraryapp.model.entity.Genre;
+import bg.libapp.libraryapp.model.entity.User;
+import bg.libapp.libraryapp.model.enums.Audit;
 import bg.libapp.libraryapp.model.mappers.BookMapper;
 import bg.libapp.libraryapp.repository.AuthorRepository;
 import bg.libapp.libraryapp.repository.BookRepository;
 import bg.libapp.libraryapp.repository.GenreRepository;
+import bg.libapp.libraryapp.repository.UserRepository;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import jakarta.transaction.Transactional;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static bg.libapp.libraryapp.model.constants.ApplicationConstants.PUBLISHER_BOOK_FIELD;
+import static bg.libapp.libraryapp.model.constants.ApplicationConstants.YEAR_BOOK_FIELD;
+
 @Service
+@Transactional
 public class BookService {
```

Comment:
Since implementing user active/inactive you should probably add some validation to each method whether the user who wants to do the action is active.

Reply:
I've implemented the isActive to be set for the isEnabled of AppUserDetails. When a user is authenticated, Spring Security checks also if the user's 'isEnabled' is true, and if it is true => the user is logged in, otherwise it returns Status 401(Unauthorized).
![image](https://github.com/dimov05/libraryapp/assets/94742531/43d25712-f319-4082-a854-62a2db303ad1)
![image](https://github.com/dimov05/libraryapp/assets/94742531/2a936af9-f209-4ceb-bb2d-ba5cc136e6c0)

Reply:
My bad, must have missed it.